lapis/job_io/htcondor.py

```diff
@@ -31,36 +32,52 @@ def htcondor_job_reader(
         "DiskUsage_RAW": 1.024 / 1024 / 1024,
     },
 ):
-    htcondor_reader = csv.DictReader(iterable, delimiter=" ", quotechar="'")
+    input_file_type = iterable.name.split(".")[-1]
+    if input_file_type == "json":
+        htcondor_reader = json.load(iterable)
+    elif input_file_type == "csv":
+        htcondor_reader = csv.DictReader(iterable, delimiter=" ", quotechar="'")
+    else:
+        print("Invalid input file type {}. Job input file can not be read.".format(
+            input_file_type))
 
-    for row in htcondor_reader:
-        if float(row[used_resource_name_mapping["walltime"]]) <= 0:
+    for entry in htcondor_reader:
+        if float(entry[used_resource_name_mapping["walltime"]]) <= 0:
             logging.getLogger("implementation").warning(
-                "removed job from htcondor import (%s)", row
+                "removed job from htcondor import (%s)", entry
             )
             continue
         resources = {}
         for key, original_key in resource_name_mapping.items():
             try:
-                resources[key] = float(row[original_key]) * unit_conversion_mapping.get(
-                    original_key, 1
-                )
+                resources[key] = float(entry[original_key]) \
+                                 * unit_conversion_mapping.get(original_key, 1)
             except ValueError:
                 pass
         used_resources = {
             "cores": (
-                (float(row["RemoteSysCpu"]) + float(row["RemoteUserCpu"]))
-                / float(row[used_resource_name_mapping["walltime"]])
+                (float(entry["RemoteSysCpu"]) + float(entry["RemoteUserCpu"]))
+                / float(entry[used_resource_name_mapping["walltime"]])
             )
             * unit_conversion_mapping.get(used_resource_name_mapping["cores"], 1)
         }
+        try:
+            inputfiles = {file["filename"]: file["usedsize"]
+                          for file in entry["Inputfiles"]}
+        except KeyError:
+            inputfiles = dict()
         for key in ["memory", "walltime", "disk"]:
             original_key = used_resource_name_mapping[key]
             used_resources[key] = float(
-                row[original_key]
+                entry[original_key]
             ) * unit_conversion_mapping.get(original_key, 1)
+        cpu_efficiency = (float(entry["RemoteSysCpu"]) +
+                          float(entry["RemoteUserCpu"])) \
+                         / float(entry[used_resource_name_mapping["walltime"]])
```

Comment:
This is duplicated code. See definition of `used_resources["cores"]` above.

Reply:
Same as above. I will remove it